packages/mui-joy/src/Button/Button.tsx

```diff
@@ -255,6 +257,26 @@ const Button = React.forwardRef(function Button(inProps, ref) {
   };
 
   const classes = useUtilityClasses(ownerState);
+
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    if (toggleButtonGroup) {
+      props.onClick?.(event);
+      toggleButtonGroup.onClick(event, props.value);
+    } else {
+      props.onClick?.(event);
+    }
```

Comment:
No need of `else` block `props.onClick`:
```suggestion
    props.onClick?.(event);
    if (toggleButtonGroup) {
      toggleButtonGroup.onClick(event, props.value);
    }
```
Also, I think we should consider `slotProps.root.onClick`.